QEfficient/transformers/models/modeling_auto.py

```diff
@@ -149,6 +152,14 @@ def from_pretrained(cls, pretrained_model_name_or_path, continuous_batching: boo
             model.generate(prompts=["Hi there!!"])
         """
 
+        num_speculative_tokens = kwargs.pop("num_speculative_tokens", None)
+        is_dlm = kwargs.pop("is_dlm", False)
+        if num_speculative_tokens is not None:
+            if not isinstance(num_speculative_tokens, int) or num_speculative_tokens<2:
```

Comment:
Hey Onkar, I agree, the second condition alone should suffice for this. 

However, just to be confident in the expectations of this approach, if we take off the first condition, then yes, it will fail if the data type is something other than `int`. 

The error message will be something like below:
```python
>>> 'a' < 2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'str' and 'int'
```

this type of error message is not as intuitive as the current log:
```python
"`num_speculative_tokens` arg should be an integer greater than 1."
```

If okay with this, please let me know and I can make the change